activestorage/app/controllers/concerns/active_storage/set_current.rb

```diff
@@ -9,7 +9,12 @@ module ActiveStorage::SetCurrent
 
   included do
     before_action do
-      ActiveStorage::Current.url_options = { protocol: request.protocol, host: request.host, port: request.port }
```

Comment:
I'm most likely missing something but do you maybe know why we need this rather than relying on the `UrlFor` module that does exactly the same. I tried digging into the git history but couldn't see the reason.

Since we are in the context of a controller we should be able to do `ActiveStorage::Current.url_options = url_options` which would also solve the script name issue as it takes care of it as well.